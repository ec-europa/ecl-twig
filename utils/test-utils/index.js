const Twig = require('twig');
const { format } = require('prettier');

const renderTwigFile = (file, options, cb) =>
  Twig.renderFile(file, options, (err, html) =>
    cb(err, format(html, { parser: 'html' }))
  );

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) =>
    Twig.renderFile(file, options, (err, html) => {
      if (err) return reject(err);

      const div = document.createElement('div');
      div.innerHTML = html.trim();

      return resolve(div.firstChild);
    })
  );

module.exports = {
  renderTwigFile,
  renderTwigFileAsNode,
};
