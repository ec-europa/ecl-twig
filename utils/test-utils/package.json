{
  "name": "@ecl-twig/test-utils",
  "version": "2.29.0",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
