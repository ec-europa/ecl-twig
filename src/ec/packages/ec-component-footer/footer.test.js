import path from 'path';
import { merge, renderTwigFileAsNode } from '@ecl-twig/test-utils';

import { backToTop, identity, sections, common } from './demo/data';

describe('EC - Footer', () => {
  const template = path.resolve(__dirname, './footer.html.twig');
  const render = params => renderTwigFileAsNode(template, params);

  describe('Corporate', () => {
    const options = { back_to_top: backToTop, sections, common };

    test('renders correctly', () => {
      expect.assertions(1);
      return expect(render(options)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(options, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(options, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });

  describe('Custom', () => {
    const options = { back_to_top: backToTop, identity, sections, common };

    test('renders correctly', () => {
      expect.assertions(1);
      return expect(render(options)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(options, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(options, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });
});
