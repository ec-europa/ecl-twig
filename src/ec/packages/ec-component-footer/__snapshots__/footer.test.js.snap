// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - Footer Corporate renders correctly 1`] = `
"<footer class=\\"ecl-footer\\">
  <a
    href=\\"#top\\"
    class=\\"ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after ecl-footer__back-to-top\\"
    ><span class=\\"ecl-link__label\\">Go to top</span>&nbsp;

    <svg
      class=\\"ecl-icon ecl-icon--fluid ecl-link__icon\\"
      focusable=\\"false\\"
      aria-hidden=\\"true\\"
    >
      <use xlink:href=\\"static/icons.svg#ui--corner-arrow\\"></use></svg
  ></a>

  <div class=\\"ecl-footer__sections\\">
    <div class=\\"ecl-container\\">
      <div class=\\"ecl-row\\">
        <section class=\\"ecl-footer__section ecl-col-12 ecl-col-md-4\\">
          <h1 class=\\"ecl-footer__section-title\\">European Commission</h1>

          <ul class=\\"ecl-footer__section-list\\">
            <li class=\\"ecl-footer__section-item\\">
              <a
                href=\\"/example\\"
                class=\\"ecl-link ecl-link--standalone ecl-footer__section-link\\"
                >Commission and its priorities</a
              >
            </li>
            <li class=\\"ecl-footer__section-item\\">
              <a
                href=\\"/example\\"
                class=\\"ecl-link ecl-link--standalone ecl-footer__section-link\\"
                >Policies, information and services</a
              >
            </li>
          </ul>
        </section>

        <section class=\\"ecl-footer__section ecl-col-12 ecl-col-md-4\\">
          <h1 class=\\"ecl-footer__section-title\\">
            Follow the European Commission
          </h1>

          <ul class=\\"ecl-footer__section-list ecl-footer__section-list--inline\\">
            <li class=\\"ecl-footer__section-item\\">
              <a
                href=\\"/example\\"
                class=\\"ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-footer__section-link\\"
                ><svg
                  class=\\"ecl-icon ecl-icon--xs ecl-link__icon\\"
                  focusable=\\"false\\"
                  aria-hidden=\\"true\\"
                >
                  <use
                    xlink:href=\\"static/icons.svg#branded--facebook\\"
                  ></use></svg
                >&nbsp;<span class=\\"ecl-link__label\\">Facebook</span></a
              >
            </li>
            <li class=\\"ecl-footer__section-item\\">
              <a
                href=\\"/example\\"
                class=\\"ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-footer__section-link\\"
                ><svg
                  class=\\"ecl-icon ecl-icon--xs ecl-link__icon\\"
                  focusable=\\"false\\"
                  aria-hidden=\\"true\\"
                >
                  <use
                    xlink:href=\\"static/icons.svg#branded--twitter\\"
                  ></use></svg
                >&nbsp;<span class=\\"ecl-link__label\\">Twitter</span></a
              >
            </li>
            <li class=\\"ecl-footer__section-item\\">
              <a
                href=\\"/example\\"
                class=\\"ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-footer__section-link\\"
                ><svg
                  class=\\"ecl-icon ecl-icon--xs ecl-link__icon\\"
                  focusable=\\"false\\"
                  aria-hidden=\\"true\\"
                >
                  <use xlink:href=\\"static/icons.svg#ui--external\\"></use></svg
                >&nbsp;<span class=\\"ecl-link__label\\"
                  >Other social networks</span
                ></a
              >
            </li>
          </ul>
        </section>

        <section class=\\"ecl-footer__section ecl-col-12 ecl-col-md-4\\">
          <h1 class=\\"ecl-footer__section-title\\">European Union</h1>

          <ul class=\\"ecl-footer__section-list\\">
            <li class=\\"ecl-footer__section-item\\">
              <a
                href=\\"/example\\"
                class=\\"ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after ecl-footer__section-link\\"
                ><span class=\\"ecl-link__label\\">European Union</span>&nbsp;

                <svg
                  class=\\"ecl-icon ecl-icon--xs ecl-link__icon\\"
                  focusable=\\"false\\"
                  aria-hidden=\\"true\\"
                >
                  <use xlink:href=\\"static/icons.svg#ui--external\\"></use></svg
              ></a>
            </li>
            <li class=\\"ecl-footer__section-item\\">
              <a
                href=\\"/example\\"
                class=\\"ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after ecl-footer__section-link\\"
                ><span class=\\"ecl-link__label\\">EU institutions</span>&nbsp;

                <svg
                  class=\\"ecl-icon ecl-icon--xs ecl-link__icon\\"
                  focusable=\\"false\\"
                  aria-hidden=\\"true\\"
                >
                  <use xlink:href=\\"static/icons.svg#ui--external\\"></use></svg
              ></a>
            </li>
          </ul>
        </section>
      </div>
    </div>
  </div>

  <section class=\\"ecl-footer__common\\">
    <div class=\\"ecl-container ecl-footer__common-container\\">
      <a
        href=\\"/example\\"
        class=\\"ecl-link ecl-link--standalone ecl-footer__common-link\\"
        >About the Commission's new web presence</a
      >

      <a
        href=\\"/example\\"
        class=\\"ecl-link ecl-link--standalone ecl-footer__common-link\\"
        >Language policy</a
      >

      <a
        href=\\"/example\\"
        class=\\"ecl-link ecl-link--standalone ecl-footer__common-link\\"
        >Resources for partners</a
      >

      <a
        href=\\"/example\\"
        class=\\"ecl-link ecl-link--standalone ecl-footer__common-link\\"
        >Cookies</a
      >

      <a
        href=\\"/example\\"
        class=\\"ecl-link ecl-link--standalone ecl-footer__common-link\\"
        >Privacy policy</a
      >

      <a
        href=\\"/example\\"
        class=\\"ecl-link ecl-link--standalone ecl-footer__common-link\\"
        >Legal notice</a
      >

      <a
        href=\\"/example\\"
        class=\\"ecl-link ecl-link--standalone ecl-footer__common-link\\"
        >Contact</a
      >
    </div>
  </section>
</footer>
"
`;

exports[`EC - Footer Custom renders correctly 1`] = `
"<footer class=\\"ecl-footer ecl-footer--custom\\">
  <a
    href=\\"#top\\"
    class=\\"ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after ecl-footer__back-to-top\\"
    ><span class=\\"ecl-link__label\\">Go to top</span>&nbsp;

    <svg
      class=\\"ecl-icon ecl-icon--fluid ecl-link__icon\\"
      focusable=\\"false\\"
      aria-hidden=\\"true\\"
    >
      <use xlink:href=\\"static/icons.svg#ui--corner-arrow\\"></use></svg
  ></a>

  <section class=\\"ecl-footer__identity\\">
    <div class=\\"ecl-container\\">
      <div class=\\"ecl-row\\">
        <div class=\\"ecl-col-12 ecl-col-md-4\\">
          <h1 class=\\"ecl-footer__identity-title\\">Site identification</h1>
        </div>

        <div class=\\"ecl-footer__identity-follow ecl-col-12 ecl-col-md-4\\">
          <span class=\\"ecl-footer__identity-label\\">
            Follow us:
          </span>

          <a
            href=\\"/example\\"
            class=\\"ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-footer__identity-link\\"
            ><svg
              class=\\"ecl-icon ecl-icon--xs ecl-link__icon\\"
              focusable=\\"false\\"
              aria-hidden=\\"true\\"
            >
              <use xlink:href=\\"static/icons.svg#branded--facebook\\"></use></svg
            >&nbsp;<span class=\\"ecl-link__label\\">Facebook</span></a
          >

          <a
            href=\\"/example\\"
            class=\\"ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-footer__identity-link\\"
            ><svg
              class=\\"ecl-icon ecl-icon--xs ecl-link__icon\\"
              focusable=\\"false\\"
              aria-hidden=\\"true\\"
            >
              <use xlink:href=\\"static/icons.svg#branded--twitter\\"></use></svg
            >&nbsp;<span class=\\"ecl-link__label\\">Twitter</span></a
          >

          <a
            href=\\"/example\\"
            class=\\"ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-footer__identity-link\\"
            ><svg
              class=\\"ecl-icon ecl-icon--xs ecl-link__icon\\"
              focusable=\\"false\\"
              aria-hidden=\\"true\\"
            >
              <use xlink:href=\\"static/icons.svg#ui--external\\"></use></svg
            >&nbsp;<span class=\\"ecl-link__label\\">Other social networks</span></a
          >
        </div>

        <div class=\\"ecl-footer__identity-info ecl-col-12 ecl-col-md-4\\">
          <a
            href=\\"/example\\"
            class=\\"ecl-link ecl-link--standalone ecl-footer__identity-link\\"
            >Contact</a
          >

          <a
            href=\\"/example\\"
            class=\\"ecl-link ecl-link--standalone ecl-footer__identity-link\\"
            >Sitemap</a
          >

          <a
            href=\\"/example\\"
            class=\\"ecl-link ecl-link--standalone ecl-footer__identity-link\\"
            >Lorem ipsum</a
          >

          <a
            href=\\"/example\\"
            class=\\"ecl-link ecl-link--standalone ecl-footer__identity-link\\"
            >Lorem ipsum dolor sit</a
          >
        </div>
      </div>
    </div>
  </section>

  <div class=\\"ecl-footer__sections\\">
    <div class=\\"ecl-container\\">
      <div class=\\"ecl-row\\">
        <section class=\\"ecl-footer__section ecl-col-12 ecl-col-md-4\\">
          <h1 class=\\"ecl-footer__section-title\\">European Commission</h1>

          <ul class=\\"ecl-footer__section-list\\">
            <li class=\\"ecl-footer__section-item\\">
              <a
                href=\\"/example\\"
                class=\\"ecl-link ecl-link--standalone ecl-footer__section-link\\"
                >Commission and its priorities</a
              >
            </li>
            <li class=\\"ecl-footer__section-item\\">
              <a
                href=\\"/example\\"
                class=\\"ecl-link ecl-link--standalone ecl-footer__section-link\\"
                >Policies, information and services</a
              >
            </li>
          </ul>
        </section>

        <section class=\\"ecl-footer__section ecl-col-12 ecl-col-md-4\\">
          <h1 class=\\"ecl-footer__section-title\\">
            Follow the European Commission
          </h1>

          <ul class=\\"ecl-footer__section-list ecl-footer__section-list--inline\\">
            <li class=\\"ecl-footer__section-item\\">
              <a
                href=\\"/example\\"
                class=\\"ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-footer__section-link\\"
                ><svg
                  class=\\"ecl-icon ecl-icon--xs ecl-link__icon\\"
                  focusable=\\"false\\"
                  aria-hidden=\\"true\\"
                >
                  <use
                    xlink:href=\\"static/icons.svg#branded--facebook\\"
                  ></use></svg
                >&nbsp;<span class=\\"ecl-link__label\\">Facebook</span></a
              >
            </li>
            <li class=\\"ecl-footer__section-item\\">
              <a
                href=\\"/example\\"
                class=\\"ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-footer__section-link\\"
                ><svg
                  class=\\"ecl-icon ecl-icon--xs ecl-link__icon\\"
                  focusable=\\"false\\"
                  aria-hidden=\\"true\\"
                >
                  <use
                    xlink:href=\\"static/icons.svg#branded--twitter\\"
                  ></use></svg
                >&nbsp;<span class=\\"ecl-link__label\\">Twitter</span></a
              >
            </li>
            <li class=\\"ecl-footer__section-item\\">
              <a
                href=\\"/example\\"
                class=\\"ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-before ecl-footer__section-link\\"
                ><svg
                  class=\\"ecl-icon ecl-icon--xs ecl-link__icon\\"
                  focusable=\\"false\\"
                  aria-hidden=\\"true\\"
                >
                  <use xlink:href=\\"static/icons.svg#ui--external\\"></use></svg
                >&nbsp;<span class=\\"ecl-link__label\\"
                  >Other social networks</span
                ></a
              >
            </li>
          </ul>
        </section>

        <section class=\\"ecl-footer__section ecl-col-12 ecl-col-md-4\\">
          <h1 class=\\"ecl-footer__section-title\\">European Union</h1>

          <ul class=\\"ecl-footer__section-list\\">
            <li class=\\"ecl-footer__section-item\\">
              <a
                href=\\"/example\\"
                class=\\"ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after ecl-footer__section-link\\"
                ><span class=\\"ecl-link__label\\">European Union</span>&nbsp;

                <svg
                  class=\\"ecl-icon ecl-icon--xs ecl-link__icon\\"
                  focusable=\\"false\\"
                  aria-hidden=\\"true\\"
                >
                  <use xlink:href=\\"static/icons.svg#ui--external\\"></use></svg
              ></a>
            </li>
            <li class=\\"ecl-footer__section-item\\">
              <a
                href=\\"/example\\"
                class=\\"ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after ecl-footer__section-link\\"
                ><span class=\\"ecl-link__label\\">EU institutions</span>&nbsp;

                <svg
                  class=\\"ecl-icon ecl-icon--xs ecl-link__icon\\"
                  focusable=\\"false\\"
                  aria-hidden=\\"true\\"
                >
                  <use xlink:href=\\"static/icons.svg#ui--external\\"></use></svg
              ></a>
            </li>
          </ul>
        </section>
      </div>
    </div>
  </div>

  <section class=\\"ecl-footer__common\\">
    <div class=\\"ecl-container ecl-footer__common-container\\">
      <a
        href=\\"/example\\"
        class=\\"ecl-link ecl-link--standalone ecl-footer__common-link\\"
        >About the Commission's new web presence</a
      >

      <a
        href=\\"/example\\"
        class=\\"ecl-link ecl-link--standalone ecl-footer__common-link\\"
        >Language policy</a
      >

      <a
        href=\\"/example\\"
        class=\\"ecl-link ecl-link--standalone ecl-footer__common-link\\"
        >Resources for partners</a
      >

      <a
        href=\\"/example\\"
        class=\\"ecl-link ecl-link--standalone ecl-footer__common-link\\"
        >Cookies</a
      >

      <a
        href=\\"/example\\"
        class=\\"ecl-link ecl-link--standalone ecl-footer__common-link\\"
        >Privacy policy</a
      >

      <a
        href=\\"/example\\"
        class=\\"ecl-link ecl-link--standalone ecl-footer__common-link\\"
        >Legal notice</a
      >

      <a
        href=\\"/example\\"
        class=\\"ecl-link ecl-link--standalone ecl-footer__common-link\\"
        >Contact</a
      >
    </div>
  </section>
</footer>
"
`;
